src/tiledb/cloud/utilities/_common.py

```diff
@@ -272,3 +274,73 @@ def reader(stream):
             stderr = stderr_future.result()
 
             return stdout, stderr
+
+
+def batch(func: Callable) -> Callable:
```

Comment:
Typing trick: To make functions using this decorator show up correctly in static analysis, you can use a TypeVar:

```python

_CT = TypeVar("_CT", bound=Callable)

def batch(func: _CT) -> _CT:
  ...
```

This specifically indicates to type checkers and static analysis tools that the full details of the type of the incoming function is the exact same as that of the returned value. (It's basically the static analysis equivalent of `functools.wraps`.) You can use this or just leave it as-is; I doubt many people will be using this in the context of a case where people would be using static analysis tools. (For instance, autocomplete and other hints in things like ipython use the function's `__doc__` and `__signature__`, which are unchanged by the change in type annotation.)

Separately:

`as_batch` might be a better name. Also, I would recommend that, rather than using it directly as a decorator, you have the function and expose the wrapped version, maybe by doing:

```python

def ingest_annotations(...) -> ...:
  ...

ingest_annotations_batch = as_batch(ingest_annotations)

# You could also do `_ingest_annotations` for the implementation
# and `ingest_annotations` for the exported version.
```

That way, the mechanism of what is actually run is separated from the policy of where to run it, which may be useful for testing or other scenarios.

Reply:
Thanks, made these changes.